### DISCUSSION TITLE
Can mold help detect ODR errors?

### DISCUSSION THREAD

**Discussion by cruisercoder** [rui314/mold#404] [Q&A]
ODR errors, such as mismatched destructors from two classes with the same name, are hard to diagnose at runtime.   Does mold offer anything to help detect this type of problem at link time?

**Comment by rui314** [comment]
We want it if we can implement it. GNU gold has `--detect-odr-violations` flag, but I don't know how it is implemented. It also looks like ASAN has a flag to detect an ODR violation. Have you tried them?
